js/walutomat.js

```diff
@@ -0,0 +1,274 @@
+            'operationType': 'MARKET_FX',
+            'sortOrder': 'DESC',
+        };
+        let market = undefined;
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+            request['currencies'] = market['id'];
+        }
+        if (since !== undefined) {
+            // TODO: Not sure if it's correct (idk if it's timestamp or DateTime string
+            request['continueFrom'] = since;
+        }
+        if (limit !== undefined) {
+            request['volume'] = limit;
+        }
+        const response = await this.privateGetAccountHistory (this.extend (request, params));
+        return this.parseTrades (response, market, since, limit);
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const body = {
+            'submitId': this.uuid (),
+            'pair': market['id'],
+            'price': price,
+            'buySell': side.toUpperCase (),
+            'volumeCurrency': market['baseId'],
+            'otherCurrency': market['quoteId'],
+            'volume': amount,
+        };
+        return await this.privatePostMarketOrders (this.extend (body, params));
+    }
+
+    async cancelOrder (id, symbol = undefined, params = {}) {
+        const request = {
+            'orderId': id,
+        };
+        return await this.privatePostMarketOrdersCloseOrderId (this.extend (request, params));
+    }
+
+    async fetchOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        // TODO: Would be nice to get parametrized API response
+        await this.loadMarkets ();
+        const market = undefined;
+        const orders = await this.privateGetMarketOrders ();
+        return this.parseOrders (orders, market, since, limit);
+    }
+
+    async fetchClosedOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        // TODO: Would be nice to have fetchOrders parametrized to get closed offers
+        return await this.fetchOrders (symbol, since, limit, params);
+    }
+
+    formatOrderBook (orderBookEntry) {
+        return [this.safeFloat (orderBookEntry, 'price'), this.safeFloat (orderBookEntry, 'baseVolume')];
+    }
+
+    parseOrder (order, market = undefined) {
+        let side = this.safeString (order, 'buySell');
+        if (side !== undefined) {
+            side = side.toLowerCase ();
+        }
+        return {
+            'info': order,
+            'id': this.safeString (order, 'orderId'),
+            'symbol': this.findSymbol (this.safeString (order, 'market')),
+            'timestamp': undefined,
+            'datetime': this.safeString (order, 'submitTs'),
+            'lastTradeTimestamp': undefined,
+            'type': 'limit',
+            'side': side,
+            'price': this.safeFloat (order, 'price'),
+            'cost': this.safeFloat (order, 'feeAmountMax'),
```

Comment:
This particular line has conflicts with the unified structure: https://github.com/ccxt/ccxt/wiki/Manual#order-structure, it puts the  `fee['cost']` inside the order's `cost`, however it's not clear where this structure is coming from, as i can't see this on their docs...

Reply:
oh, ok, i think it's this one: https://api.walutomat.pl/

Reply:
Yeah, indeed it v1 API docs is this URL. Do I need to make some changes than?

Reply:
Done ;)